fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedConnector.java

```diff
@@ -56,18 +60,24 @@ public UnifiedConnector(ConnectorContext context) {
         derivedProperties.putAll(context.getProperties());
         derivedProperties.put(HIVE_METASTORE_TYPE, metastoreType);
         derivedProperties.put(ICEBERG_CATALOG_TYPE, metastoreType);
+        derivedProperties.put(PAIMON_CATALOG_TYPE, metastoreType);
         derivedProperties.put(KUDU_CATALOG_TYPE, metastoreType);
 
         ConnectorContext derivedContext = new ConnectorContext(context.getCatalogName(), context.getType(),
                 derivedProperties.build());
 
-        connectorMap = ImmutableMap.of(
-                HIVE, new HiveConnector(derivedContext),
-                ICEBERG, new IcebergConnector(derivedContext),
-                HUDI, new HudiConnector(derivedContext),
-                DELTALAKE, new DeltaLakeConnector(derivedContext),
-                KUDU, new KuduConnector(derivedContext)
-        );
+        ImmutableMap.Builder<Table.TableType, Connector> builder = ImmutableMap.builder();
+
+        builder.put(HIVE, new HiveConnector(derivedContext))
+                .put(ICEBERG, new IcebergConnector(derivedContext))
+                .put(HUDI, new HudiConnector(derivedContext))
+                .put(DELTALAKE, new DeltaLakeConnector(derivedContext))
+                .put(KUDU, new KuduConnector(derivedContext));
+        boolean containsPaimon = null != context.getProperties().get(PAIMON_CATALOG_WAREHOUSE);
```

Comment:
could you move this part to `CatalogAnalyzer` ?

Reply:
Seems hard cause it's for unified catalog only. Any ideas?